devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -492,6 +496,10 @@ private void handleAutoCompile() throws MojoExecutionException {
         boolean testCompileNeeded = true;
         boolean prepareNeeded = true;
         boolean prepareTestsNeeded = true;
+
+        String jandexGoalPhase = getGoalPhaseOrNull(ORG_JBOSS_JANDEX, JANDEX_MAVEN_PLUGIN, "jandex", "process-classes");
```

Comment:
So I had to add the defaut phase (last parameter) to cover the very common case of a missing `phase` element.